includes/class-wc-order-stats.php

```diff
@@ -0,0 +1,420 @@
+		$oldest = wc_get_orders( array(
+			'limit'   => 1,
+			'orderby' => 'date',
+			'order'   => 'ASC',
+			'status'  => self::get_report_order_statuses(),
+			'type'    => 'shop_order',
+		) );
+
+		$oldest = $oldest ? reset( $oldest ) : false;
+		if ( ! $oldest ) {
+			return;
+		}
+
+		$start_time = strtotime( $oldest->get_date_created()->format( 'Y-m-d\TH:00:00O' ) );
+		$end_time = $start_time + HOUR_IN_SECONDS;
+
+		while ( $end_time < time() ) {
+			self::$background_process->push_to_queue( $start_time );
+			$start_time = $end_time;
+			$end_time = $start_time + HOUR_IN_SECONDS;
+		}
+
+		self::$background_process->save();
+	}
+
+	/**
+	 * Queue a background process that will update the database with stats info from the last hour.
+	 */
+	public function queue_update_recent_orders() {
+		// Populate the stats information for the previous hour.
+		$last_hour = strtotime( date( 'Y-m-d H:00:00' ) ) - HOUR_IN_SECONDS;
+		self::$background_process->push_to_queue( $last_hour );
+		self::$background_process->save();
+	}
+
+	/**
+	 * Schedule a recalculation for an order's time when the order gets updated.
+	 * This schedules for the order's current time and for the time the order used to be in if necessary.
+	 */
+	public function queue_update_modified_orders( $order ) {
+		$new_time = strtotime( $order->get_date_created()->format( 'Y-m-d\TH:00:00O' ) );
+		$old_time = false;
+		$new_status = $order->get_status();
+		$old_status = false;
+
+		if ( $order->get_id() ) {
+			$old_order = wc_get_order( $order->get_id() );
+			$old_time = strtotime( $old_order->get_date_created()->format( 'Y-m-d\TH:00:00O' ) );
+			$old_status = $old_order->get_status();
+		}
+
+		$order_statuses = self::get_report_order_statuses();
+		if ( in_array( $new_status, $order_statuses, true ) || in_array( $old_status, $order_statuses, true ) ) {
+			self::$background_process->push_to_queue( $new_time );
+
+			if ( $old_time && $new_time !== $old_time ) {
+				self::$background_process->push_to_queue( $old_time );
+			}
+
+			self::$background_process->save();
+		}
+	}
+
+	/**
+	 * Kick off any scheduled data recalculations.
+	 */
+	public function dispatch_recalculator() {
+		self::$background_process->dispatch();
+	}
+
+	/**
+	 * Get stats summary information for orders between two time frames.
+	 *
+	 * @param int $start_time Timestamp.
+	 * @param int $end_time Timestamp.
+	 * @return Array of stats.
+	 */
+	public static function summarize_orders( $start_time, $end_time ) {
+		$summary = array(
+			'num_orders'            => 0,
+			'num_items_sold'        => 0,
+			'orders_gross_total'    => 0.0,
+			'orders_coupon_total'   => 0.0,
+			'orders_refund_total'   => 0.0,
+			'orders_tax_total'      => 0.0,
+			'orders_shipping_total' => 0.0,
+			'orders_net_total'      => 0.0,
+		);
+
+		$orders = wc_get_orders( array(
+			'limit'        => -1,
+			'type'         => 'shop_order',
+			'orderby'      => 'date',
```

Comment:
I think we can significantly speed up this query if we can remove the order by clause as it will remove the need to do a filesort, and it seems to me that the order here doesn't matter. The problem is that if we don't pass the `orderby` and `order` arguments, WP will add it anyway (saying this off the top of my head, worth checking).

Reply:
Agree with @rodrigoprimo here, also since we are summing the data it does not matter in what order it is returned as long as it returns the orders for that hour period.